### PR TITLE
Simplifying code structure and cleaning script

### PR DESCRIPTION
**Pull request by An78toi13ne06**

### CODE REVIEW

File: Invoke-HardeningKitty.ps1 [hunk 1]
```diff
@@ -492,6 +492,22 @@
         Return $AccountSid
     }
 
+    function Write-NotAdminError {
+        [CmdletBinding()]
+        param (
+            [Int]
```
Comment (0x6d69636b) [inline Invoke-HardeningKitty.ps1]: The CIS benchmark list for example has an ID like _18.9.108.4.2.2_, therefore we should use `[String]` for the ID as well

File: Invoke-HardeningKitty.ps1 [hunk 2]
```diff
@@ -537,7 +553,7 @@
     $StatsMedium = 0
     $StatsHigh = 0
     $StatsTotal = 0
-    $StatsError = 0
+    $script:StatsError = 0
```
Comment (0x6d69636b) [inline Invoke-HardeningKitty.ps1]: Could you use a capital letter for the `$Script` variable, please?

File: Invoke-HardeningKitty.ps1 [hunk 3]
```diff
@@ -1,2269 +1,2266 @@
...
+        $Message = "ID "+$FindingID+", "+$FindingName+", Method "+$FindingMethod+" requires admin priviliges. Test skipped."
+        Write-ProtocolEntry -Text $Message -LogLevel "Error"
+    }
+
+    function Write-BinaryError {
+        [CmdletBinding()]
+        param (
+            [String]
+            $Binary,
+            [Int]
```
Comment (0x6d69636b) [inline Invoke-HardeningKitty.ps1:516]: Again, we should use `[String]` for the ID

File: Invoke-HardeningKitty.ps1 [hunk 4]
```diff
@@ -1,2289 +1,2262 @@
...
+                    }
+                    Else {
+                        $ResultOutputList = $ResultOutputRaw.ToString().split("=").Trim()
+                        $Result = $ResultOutputList[1] -Replace "\*",""
+                        $Result = $Result -Replace ",",";"
+                    }
+
+                } catch {
+                    # If secedit did not work, throw an error instead of using the DefaultValue
+                    $StatsError++
```
Comment (0x6d69636b) [inline Invoke-HardeningKitty.ps1:992]: This should be changed to $Script:StatsError as well

File: Invoke-HardeningKitty.ps1 [hunk 5]
```diff
@@ -1,2289 +1,2262 @@
...
+            $HardeningKittyScore = ([int] $ScoreAchived / [int] $ScoreTotal) * 5 + 1
+        }        
+        $HardeningKittyScoreRounded = [math]::round($HardeningKittyScore,2)
+
+        # Overwrite HardeningKitty Score if no finding is passed
+        If ($StatsPassed -eq 0 ) {
+            $HardeningKittyScoreRounded = 1.00
+        }
+
+        If ($StatsError -gt 0) {
```
Comment (0x6d69636b) [inline Invoke-HardeningKitty.ps1:2255]: Here we should replace the variable to $Script:StatsError as well, and in the output below

**Review by 0x6d69636b — CHANGES_REQUESTED** [review]
Thank you very much for your tremendous work! :heart_eyes_cat: 
I added a few comments, please resolve them and after that I'll merge it

**Comment by An78toi13ne06** [comment]
Changes made ^^
Don't know how to show it with github but each change asked has been resolved.

**Comment by 0x6d69636b** [comment]
I guess you just can click on the "Resolve conversation" button. But it doesn't matter if you changed it anyway ;-)

**Pull request merged**